Library/Homebrew/extend/ENV/std.rb

```diff
@@ -303,6 +303,10 @@ def libstdcxx
     end
   end
 
+  def java_cache
+    append "_JAVA_OPTIONS", "-Duser.home=#{HOMEBREW_CACHE}/java_cache"
+  end
```

Comment:
Shouldn’t we put this in `ENV/shared.rb` since it’s included in both `std` and `super`?